partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -136,33 +137,54 @@ class PartiQLSchemaInferencerTests {
     fun testSubqueries(tc: TestCase) = runTest(tc)
 
     companion object {
+        val inputStream = this::class.java.getResourceAsStream("/resource_path.txt")!!
 
-        private val root = this::class.java.getResource("/catalogs/default")!!.toURI().toPath().pathString
+        val catalogProvider = MemoryCatalog.Provider().also {
+            val map = mutableMapOf<String, MutableList<Pair<String, StaticType>>>()
+            inputStream.reader().readLines().forEach { path ->
+                if (path.startsWith("catalogs/default")) {
+                    val schema = this::class.java.getResourceAsStream("/$path")!!
+                    val ion = loadSingleElement(schema.reader().readText())
+                    val staticType = ion.toStaticType()
+                    val steps = path.split('/').drop(2) // drop the catalogs/default
+                    val catalogName = steps.first()
+                    val subPath = steps
+                        .drop(1)
+                        .joinToString(".") { it.lowercase() }
+                        .let {
+                            it.substring(0, it.length - 4)
+                        }
+                    if (map.containsKey(catalogName)) {
+                        map[catalogName]!!.add(subPath to staticType)
+                    } else {
+                        map[catalogName] = mutableListOf(subPath to staticType)
+                    }
+                }
+            }
+            map.forEach { (k: String, v: MutableList<Pair<String, StaticType>>) ->
+                it[k] = MemoryCatalog.of(*v.toTypedArray())
+            }
+        }
```

Comment:
I see, rather than porting the catalogs from the .ion files you create the mem catalog from them. This works, but would be nice to clean up the resources in a later PR. Again, these are testing concerns so nothing pressing